mrcnn/scripts/train.py

```diff
@@ -0,0 +1,190 @@
+                        # these lines have been commented out due to invalid test data file name
+                        class_id = int(mask_file.split('__')[1][:-4])
+                        class_ids.append(class_id)
+                        mask_ = cv2.imread(path + MASK_PATH + mask_file, 0)
+                        mask_ = np.where(mask_ > 128, 1, 0)
+                        # Add mask only if its area is larger than one pixel
+                        if np.sum(mask_) >= 1:
+                            mask.append(np.squeeze(mask_))
+            mask = np.stack(mask, axis=-1)
+        return mask.astype(np.uint8), class_ids.astype(np.int8)
+
+
+class ClomaskTrain(object):
+    """
+        Training class for clomask dataset.
+    """
+
+    def __init__(self):
+        self.init_weights = 'coco'
+        self.config = ClomaskConfig()
+        self.learning_rate_one = 1e-4
+        self.learning_rate_two = 1e-5
+        self.learning_rate_three = 1e-6
+        self.train_data = None
+        self.val_data = None
+
+    def prepare_dataset(self):
+
+        train_list, test_list = os.listdir(TRAIN_PATH), os.listdir(TEST_PATH)
+    #     train_list, val_list = train_test_split(os.listdir(train_path), test_size=0.1,
+    #                                             random_state=2019)
+
+    # Use this for explode the training data for use in augmentation.
+    # train_list = np.repeat(train_list,5)
+
+        # initialize training dataset
+        train_data = ClomaskDataset()
+        train_data.load_shapes(train_list, TRAIN_PATH)
+        train_data.prepare()
+
+        # initialize validation dataset
+        validation_data = ClomaskDataset()
+        validation_data.load_shapes(test_list, TEST_PATH)
+        validation_data.prepare()
+        # Create model configuration in training mode
+
+        self.config.STEPS_PER_EPOCH = len(train_list)//self.config.BATCH_SIZE
+        self.config.VALIDATION_STEPS = len(test_list)//self.config.BATCH_SIZE
+
+        self.train_data = train_data
+        self.val_data = validation_data
+
+    def weight_initialize(self):
+        """
+            Loading Model weights to start training with
+        """
+        self.model = modellib.MaskRCNN(mode="training", config=self.config, model_dir=MODEL_DIR)
+        if self.init_weights == "imagenet":
+            weights_path = self.model.get_imagenet_weights()
+            self.model.load_weights(weights_path, by_name=True, exclude=["mrcnn_class_logits", "mrcnn_bbox_fc",
+                                                                   "mrcnn_bbox", "mrcnn_mask"])
+        elif self.init_weights == "coco":
+            weights_path = COCO_PATH
+            self.model.load_weights(weights_path, by_name=True, exclude=["mrcnn_class_logits", "mrcnn_bbox_fc",
+                                                                   "mrcnn_bbox", "mrcnn_mask"])
+
+    def model_train(self):
+
+        augmentation = iaa.Sequential([
```

Comment:
Keep this line number and file in mind whenever Yipeng asks if we perform any image augmentation during training.